docs/statusreport-controller.md

```diff
@@ -3,113 +3,113 @@
+    update_status(Update Snapshot status accordingly)
+    continue_processing_tests(Controller continues processing)
+
+
+%% Node connections
+    predicate                    ---->    |"EnsureSnapshotFinishedAllTests()"|get_required_scenarios
+    get_required_scenarios        --->    parse_snapshot_status
+    parse_snapshot_status         --->    check_finished_tests
+    check_finished_tests      --Yes-->    check_supersede
+    check_finished_tests       --No-->    continue_processing_tests
+    check_supersede           --Yes-->    create_snapshot
+    check_supersede            --No-->    check_passed_tests
+    create_snapshot               --->    update_status
+    check_passed_tests        --Yes-->    update_status
+    check_passed_tests         --No-->    update_status
+    update_status                ---->    continue_processing_tests
+
+%%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureSnapshotTestStatusReportedToGitProvider() function
+
+%% Node definitions
+    ensure(Process further if: Snapshot has label <br>pac.test.appstudio.openshift.io/git-provider:github <br>defined)
+    get_annotation_value(Get integration test status from annotation <br>test.appstudio.openshift.io/status <br>from Snapshot)
+
+    detect_git_provider{Detect git provider}
+
+    collect_commit_info_gh(Collect commit owner, repo and SHA from Snapshot)
+
+    is_installation_defined{Is annotation <br>pac.test.appstudio.openshift.io/installation-id <br>defined?}
+
+    create_appInstallation_token(Create github application installation token)
+    get_all_checkRuns_from_gh(Get all checkruns from github <br>according to <br>commit owner, repo and SHA)
+    create_checkRunAdapter(Create checkRun adapter according to <br>commit owner, repo, SHA <br>and integration test status)
+    does_checkRun_exist{Does checkRun exist <br>on github already?}
+    create_new_checkRun_on_gh(Create new checkrun on github)
+    is_checkRun_update_needed{Does existing checkRun <br>have different text?}
+    update_existing_checkRun_on_gh(Update existing checkRun on github)
+
+    set_oAuth_token(Get token from Snapshot and set oAuth token)
+    get_all_commitStatuses_from_gh(Get all commitStatuses from github <br>according to commit owner, repo and SHA)
+    create_commitStatusAdapter(Create commitStatusAdapter according to <br>commit owner, repo, SHA <br>and integration test status)
+    does_commitStatus_exist{Does commitStatus exist <br>on github already?}
+    create_new_commitStatus_on_gh(Create new commitStatus on github)
+    does_comment_exist(Does a comment exist for snapshot and scenario?)
+    update_existing_comment(Update the existing comment for <br>snapshot and scenario</br>)
+    create_new_comment(Create a new comment for <br>snapshot and scenario</br>)
+
+    collect_commit_info_gl(Collect commit projectID, repo-url and SHA from Snapshot)
+    report_commit_status_gl(Create/update commitStatus on Gitlab)
+
+    is_test_final{Is<br> the test in it's <br>final state?}
```

Comment:
I think we'll also need to express that we loop through all scenarios in the Snapshot's status and then mention this condition where we check if the test is in its final state.

Reply:
I changed it to try to reflect the loop but I'm not sure if the changes adequately illustrate the loop and if there should be an additional connection or note for the termination of the loop